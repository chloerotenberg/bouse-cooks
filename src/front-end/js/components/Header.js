import React from 'react';
import { Link } from 'react-router-dom';
import { instaLink } from '../constants';

export default class Header extends React.Component {
  render() {
    return (
      <div className="header">
        <Link to="/">bouse</Link>
        {' / '}
        <Link to="/cookbot/week">cookbot</Link>
        {' / '}
        <a
          className="external"
          target="_blank"
          rel="noopener noreferrer"
          href={instaLink}
        >
          insta
        </a>
      </div>
    );
  }
}
