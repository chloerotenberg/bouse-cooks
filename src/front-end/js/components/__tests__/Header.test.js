import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('Header component', () => {
  it('matches snapshot', () => {
    const links = [
      {
        to: '/test',
        text: 'test',
        external: false
      },
      {
        to: 'http://test.example.com',
        text: 'test example',
        isExternal: true
      }
    ];
    const wrapper = shallow(<Header links={links} />);
    expect(wrapper).toMatchSnapshot();
  });
});
