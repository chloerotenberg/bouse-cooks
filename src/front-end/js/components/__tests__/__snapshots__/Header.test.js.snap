// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div
  className="header"
>
  <Link
    replace={false}
    to="/"
  >
    bouse
  </Link>
   / 
  <Link
    replace={false}
    to="/cookbot/week"
  >
    cookbot
  </Link>
   / 
  <a
    className="external"
    href="https://www.instagram.com/bousecoop/"
    rel="noopener noreferrer"
    target="_blank"
  >
    insta
  </a>
</div>
`;
